rust/windows-client/src-tauri/src/client/gui.rs

```diff
@@ -485,9 +491,24 @@ async fn run_controller(
                         controller.tunnel_ready = false;
                         if let Some(mut session) = controller.session.take() {
                             tracing::debug!("disconnecting connlib");
+                            // This is probably redundant since connlib shuts itself down if it's disconnected.
                             session.connlib.disconnect(None);
                         }
-                    },
+                        controller.refresh_system_tray_menu()?;
+                    }
+                    Req::DisconnectedTokenExpired | Req::SignOut => {
+                        tracing::debug!("Token expired or user signed out");
+                        controller.auth.sign_out()?;
+                        controller.tunnel_ready = false;
+                        if let Some(mut session) = controller.session.take() {
+                            tracing::debug!("disconnecting connlib");
+                            session.connlib.disconnect(None);
```

Comment:
No need to disconnect connlib, when it detects an expired token it just closes itself.

Reply:
Okay. It would be hard to fix since I'm using that same match arm for user-initiated sign-outs, too. I just added a comment there in case the code moves in the future.